### PR TITLE
Combine title changes in the same job in Resume

### LINKED ISSUE #125: Combine same job change titles in Resumes
**Is your feature request related to a problem? Please describe.**
Right now if you changed your title while workin on a company you need to create multiple `work` items and they will be displayed independently.

**Describe the solution you'd like**
Use a similar approach as LinkedIn and show them grouped together as a single work experience with a timeline of titles.

**Additional context**
Examples:
- https://sergiodxa.com/cv/
- https://github.com/sergiodxa/personal-site/blob/master/resume.json#L31-L50

### PR DESCRIPTION
**Pull request by cristianbgp**
Resolves #125

### CODE REVIEW

File: src/components/resume.tsx [hunk 1]
```diff
@@ -70,50 +100,88 @@ function Basics(props: Basics) {
 function WorkExperience(props: { experiencies: Work[] }) {
   if (props.experiencies.length === 0) return null;
 
+  let groupedExperiences = props.experiencies.reduce(
+    (accumulator: Work[][], current) => {
```
Comment (sergiodxa) [inline src/components/resume.tsx]:
```suggestion
    (works: Work[][], work) => {
```

File: src/components/resume.tsx [hunk 2]
```diff
@@ -70,50 +100,88 @@ function Basics(props: Basics) {
...
+      if (
+        accumulator.length &&
```
Comment (sergiodxa) [inline src/components/resume.tsx]:
```suggestion
        works.length &&
```

File: src/components/resume.tsx [hunk 3]
```diff
@@ -70,50 +100,88 @@ function Basics(props: Basics) {
...
+        current.company === accumulator[accumulator.length - 1][0].company
```
Comment (sergiodxa) [inline src/components/resume.tsx]:
```suggestion
        work.company === works[works.length - 1][0].company
```

File: src/components/resume.tsx [hunk 4]
```diff
@@ -70,50 +100,88 @@ function Basics(props: Basics) {
...
+      ) {
+        accumulator[accumulator.length - 1].push(current);
```
Comment (sergiodxa) [inline src/components/resume.tsx]:
```suggestion
        works[works.length - 1].push(work);
```

File: src/components/resume.tsx [hunk 5]
```diff
@@ -70,50 +100,88 @@ function Basics(props: Basics) {
...
+      } else {
+        accumulator.push([current]);
```
Comment (sergiodxa) [inline src/components/resume.tsx]:
```suggestion
        works.push([work]);
```

File: src/components/resume.tsx [hunk 6]
```diff
@@ -70,50 +100,88 @@ function Basics(props: Basics) {
...
+      }
+      return accumulator;
```
Comment (sergiodxa) [inline src/components/resume.tsx]:
```suggestion
      return works;
```

File: src/components/resume.tsx (hunk 1 above)
Comment (cristianbgp) [reply src/components/resume.tsx]: Thanks, I'll be uploading that with my new changes

**Pull request merged**